### PR TITLE
External tileset updates and documentation

### PR DESCRIPTION
**Pull request by chaosvolt**
#### Summary

SUMMARY: Interface "Fix an incorrect ID in external tileset, add a few more sprites and documentation tracking additions"

#### Purpose of change

This fixes a mistake I made last update with the external tileset, adds support for two new weapons previously added, and adds a document laying out the reason behind the external_tileset folder and an overview of all things covered by its sprites.

#### Describe the solution

1. Fixed sprite overrides for composite greatbow citing wrong item ID.
2. Added sprites for katar and sai.
3. Added documentation, which covers all content currently warranting a new or overriding sprite.

#### Describe alternatives you've considered

Putting documentation file in the doc folder instead. Main issue with that is this helps make it easier to remember to update the listed changes in said file.

#### Testing

1. Checked affected tile config for syntax and lint errors.
2. Loaded up in compiled test build, composite greatbow doesn't look like a compound bow in UDP, sai and katar have sprites.

### CODE REVIEW

File: data/json/external_tileset/External Tileset Info.md
Comment (scarf005) [inline data/json/external_tileset/External Tileset Info.md:1]: i know it's ton of work but could you also link relevent PRs to `specific to BN`?

**Review by scarf005 — APPROVED** [review]
image

File: data/json/external_tileset/External Tileset Info.md
Comment (chaosvolt) [reply data/json/external_tileset/External Tileset Info.md:1]: I guess I could, gimme a bit then.

Comment (chaosvolt) [reply data/json/external_tileset/External Tileset Info.md:1]: And implemented.

**Review by scarf005 — APPROVED** [review]

**Pull request merged**